registry/storage/driver/swift/swift.go

```diff
@@ -537,14 +544,21 @@ func (d *driver) Delete(ctx context.Context, path string) error {
 		for i, obj := range objects {
 			filenames[i] = obj.Name
 		}
-		_, err = d.Conn.BulkDelete(d.Container, filenames)
-		// Don't fail on ObjectNotFound because eventual consistency
-		// makes this situation normal.
-		if err != nil && err != swift.Forbidden && err != swift.ObjectNotFound {
-			if err == swift.ContainerNotFound {
-				return storagedriver.PathNotFoundError{Path: path}
+
+		for offset := 0; offset < len(filenames); offset += d.BulkDeleteMaxDeletes {
+			chunkSize := d.BulkDeleteMaxDeletes
+			if offset+chunkSize > len(filenames) {
+				chunkSize = len(filenames) - offset
+			}
+			_, err := d.Conn.BulkDelete(d.Container, filenames[offset:offset+chunkSize])
```

Comment:
added > 0 chunk size validation to stop an infinite loop, however not sure it can run out of bounds?

Similar code is used in the module for Swift chunked uploads in `func (sw *segmentWriter) Write(p []byte) (int, error)` and apart from the 0 situation, seems to cope with <, = and > chunk vs array size: https://play.golang.org/p/dyhw2vM08z